src/components/Icon/index.tsx

```diff
@@ -0,0 +1,8 @@
+export interface IconProps {
+  color?: string;
+  size?: 'xs' | 's' | 'm' | 'l' | 'xl';
+  customSize?: number;
+}
+
+type Icon = (props: IconProps) => JSX.Element;
```

Comment:
is there any convention to identify that this is a type vs a component? I thought in the other file that this was importing a component. cc @sfrieson

Reply:
The reason is a type and not a function (component) is because it is an abstraction. We could have created a class component and then extend it on each icon.

Reply:
Right. My concern is mostly convention based but I think this is fine. Contextually it makes sense now.